cpp/src/arrow/io/memory.cc

```diff
@@ -33,16 +33,67 @@
 namespace arrow {
 namespace io {
 
+void SerialMemcopy::memcopy(uint8_t* dst, const uint8_t* src, uint64_t nbytes) {
+  std::memcpy(dst, src, nbytes);
+}
+
+void ParallelMemcopy::memcopy(uint8_t* dst, const uint8_t* src, uint64_t nbytes) {
+  if (nbytes >= BYTES_IN_MB) {
+    memcopy_aligned(dst, src, nbytes, block_size_);
+  } else {
+    memcpy(dst, src, nbytes);
```

Comment:
I'm indifferent between `std::memcpy` and `memcpy` as long as we use one consistently